### PR TITLE
Add decode/encode for sub/suback/unsub/unsuback

### PR DESCRIPTION
**Pull request by wanghaEMQ**

### CODE REVIEW

File: demo/mqttv5/mqttv5_client.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Author: eeff <eeff at eeff dot dev>
+// Author: wangha <wanghamax at gmail dot com>
```
Comment (JaylinYu) [inline demo/mqttv5/mqttv5_client.c:1]: hage v5

File: src/mqtt/transport/tcp/mqtt_tcp.c [hunk 2]
```diff
@@ -882,7 +882,7 @@ mqtt_tcptran_pipe_start(
 
 	mqtt_version = nni_mqtt_msg_get_connect_proto_version(connmsg);
 
-	if (mqtt_version != 4 || mqtt_version != 5) {
+	if (mqtt_version != 4 && mqtt_version != 5) {
```
Comment (JaylinYu) [inline src/mqtt/transport/tcp/mqtt_tcp.c:885]: my bad

**Review by JaylinYu — APPROVED** [review]

File: demo/mqttv5/mqttv5_client.c (hunk 1 above)
Comment (wanghaEMQ) [reply demo/mqttv5/mqttv5_client.c:1]: who is eeff ?

File: src/mqtt/transport/tcp/mqtt_tcp.c (hunk 2 above)
Comment (wanghaEMQ) [reply src/mqtt/transport/tcp/mqtt_tcp.c:885]: mine too...

**Pull request merged**